Source/AppLovinAdapter.swift

```diff
@@ -48,25 +51,34 @@ final class AppLovinAdapter: PartnerAdapter {
             log(.setUpFailed(error))
             return completion(error)
         }
-        guard let sdk = ALSdk.shared(withKey: sdkKey) else {
-            let error = error(.initializationFailureInvalidCredentials, description: "Invalid \(String.sdkKey)")
-            log(.setUpFailed(error))
-            return completion(error)
-        }
-        Self.sdk = sdk
-
-        sdk.mediationProvider = "Chartboost"
-        sdk.initializeSdk { _ in
-            if sdk.isInitialized {
-                self.log(.setUpSucceded)
-                completion(nil)
-            }
+        
+        let initConfig = ALSdkInitializationConfiguration(sdkKey: sdkKey) {builder in
+            builder.mediationProvider = "Chartboost"
+            if AppLovinAdapterConfiguration.testMode {
+                let idfa = ASIdentifierManager.shared().advertisingIdentifier
+                    if idfa.uuidString != "00000000-0000-0000-0000-000000000000" {
+                        builder.testDeviceAdvertisingIdentifiers = [idfa.uuidString]
+                    }
+                }
             else {
```

Comment:
can we fix the indentation here?
```suggestion
                if idfa.uuidString != "00000000-0000-0000-0000-000000000000" {
                    builder.testDeviceAdvertisingIdentifiers = [idfa.uuidString]
                }
            } else {
```